mne/viz/backends/renderer.py

```diff
@@ -0,0 +1,51 @@
+import sys
+import os
+import importlib
+
+default_3D_backend = 'mayavi'
+
+
+try:
+    if MNE_3D_BACKEND is None:
+        MNE_3D_BACKEND = os.environ.get('MNE_3D_BACKEND',
+                                        default_3D_backend)
+except NameError:
+    MNE_3D_BACKEND = os.environ.get('MNE_3D_BACKEND', default_3D_backend)
+
+if MNE_3D_BACKEND == 'mayavi':
+    from ._pysurfer_mayavi import *
+else:
+    import warnings
+    warnings.warn('MNE_3D_BACKEND should be "mayavi" : '
+                  '{} was given.'.format(MNE_3D_BACKEND))
+
+sys.stderr.write('Using {} backend.\n'.format(MNE_3D_BACKEND))
+
+
+def set_3D_backend(backend_name):
+    """Set the backend for MNE.
+
+    The backend will be set as specified and operations will use
+    that backend
+
+    Parameters
+    ----------
+    backend_name : {'mlab'}, default is 'mlab'
+    """
+    global MNE_3D_BACKEND
```

Comment:
tensorly also uses a global?

thinking out loud maybe a context manage at some point would be nice to have:

```
with mne.3d_backend('mayavi'):
    ....
```

Reply:
tensorly uses global in `set_backend()` and `get_backend()` as well.

Reply:
Can you add a smoke test that you can do `set_3d_backend('mayavi')` followed by `set_3d_backend('mayavi')` without it breaking things (do some simple plot afterward)?

And I think that `set_3d_backend('foo')` should be a `ValueError` (rather than `warn`), and `set_3d_backend(1)` should be a TypeError (checked with `mne.utils._validate_type`) and that these degenerate conditions should be checked with `with pytest.raises(..., match='...'):` in a new `mne/viz/backends/tests/test_renderer.py` file.